rust/src/transformations/make_stable_expr/expr_binary/mod.rs

```diff
@@ -52,13 +52,19 @@ where
         return fallible!(MakeTransformation, "unsupported operator: {:?}. Only binary operations that emit booleans are currently supported.", op);
     }
 
-    let mut data_column =
-        SeriesDomain::new(expr_output_name(&expr)?, AtomDomain::<bool>::default());
+    let left_series = &t_left.output_domain.column;
+    let right_series = &t_right.output_domain.column;
 
-    let left_nullable = t_left.output_domain.column.nullable;
-    let right_nullable = t_right.output_domain.column.nullable;
+    if matches!(left_series.dtype(), DataType::Categorical(_, _)) {
+        return fallible!(MakeTransformation, "{} cannot be applied to categorical data, because it may trigger a data-dependent CategoricalRemappingWarning in Polars", op);
+    }
+    if matches!(right_series.dtype(), DataType::Categorical(_, _)) {
+        return fallible!(MakeTransformation, "{} cannot be applied to categorical data, because it may trigger a data-dependent CategoricalRemappingWarning in Polars", op);
+    }
```

Comment:
Would it make sense to combine these two with an `||` so the return doesn't need to be repeated?

Hoping that a python user wouldn't be able to hit these errors?